File: backend/app/api/v1/endpoints/maps.py
"""
PRAGMA — MAPs Endpoints

Owner: Diyasha (Backend APIs)
Milestone: M2

Endpoints:
  GET   /maps                   — List all MAPs (filter: status, department, priority)
  GET   /maps/{id}              — Get single MAP with approval history
  PATCH /maps/{id}/status       — Department marks MAP as In Progress / Completed
"""

from fastapi import APIRouter

router = APIRouter()

# TODO (M2): Implement endpoints — see docs/api-reference.md for full spec

File: frontend/src/api/maps.js
/**
 * PRAGMA — MAPs API
 * Owner: Ashwin + Diyasha — M2/M3
 */

import api from '../services/api'

export const getMAPs = (filters = {}) =>
  api.get('/maps', { params: filters }).then((r) => r.data)

export const getMAPById = (id) =>
  api.get(`/maps/${id}`).then((r) => r.data)

export const updateMAPStatus = (id, status) =>
  api.patch(`/maps/${id}/status`, { status }).then((r) => r.data)
